File: java/src/test/java/io/github/securekit/PasswordHasherTest.java
package io.github.securekit;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class PasswordHasherTest {

    @Test
    void roundTrip() {
        PasswordHasher hasher = new PasswordHasher();
        String hash = hasher.hash("correct horse battery staple");
        assertTrue(hasher.verify("correct horse battery staple", hash));
    }

    @Test
    void rejectsWrongPassword() {
        PasswordHasher hasher = new PasswordHasher();
        String hash = hasher.hash("correct-password");
        assertFalse(hasher.verify("wrong-password", hash));
    }

    @Test
    void needsRehashDetectsWeakerParams() {
        PasswordHasher weak = new PasswordHasher(8 * 1024, 1, 1, 16, 32);
        String hash = weak.hash("password");

        PasswordHasher strong = new PasswordHasher();
        assertTrue(strong.needsRehash(hash));
    }

    @Test
    void rejectsEmptyPassword() {
        assertThrows(IllegalArgumentException.class, () -> new PasswordHasher().hash(""));
    }
}

File: php/tests/PasswordHasherTest.php
<?php

declare(strict_types=1);

namespace SecureKit\Tests;

use PHPUnit\Framework\TestCase;
use SecureKit\PasswordHasher;

final class PasswordHasherTest extends TestCase
{
    public function testRoundTrip(): void
    {
        $hasher = new PasswordHasher();
        $hash = $hasher->hash('correct horse battery staple');
        $this->assertTrue($hasher->verify('correct horse battery staple', $hash));
    }

    public function testRejectsWrongPassword(): void
    {
        $hasher = new PasswordHasher();
        $hash = $hasher->hash('correct-password');
        $this->assertFalse($hasher->verify('wrong-password', $hash));
    }

    public function testNeedsRehashDetectsWeakerParams(): void
    {
        $weak = new PasswordHasher(memoryCostKiB: 8192, timeCost: 1, parallelism: 1);
        $hash = $weak->hash('password');

        $strong = new PasswordHasher();
        $this->assertTrue($strong->needsRehash($hash));
    }

    public function testRejectsEmptyPassword(): void
    {
        $this->expectException(\InvalidArgumentException::class);
        (new PasswordHasher())->hash('');
    }
}
